src/action_handle_hashtags.py

```diff
@@ -0,0 +1,143 @@
+from functools import partial
+from random import shuffle
+import random
+
+import uiautomator
+
+from src.storage import FollowingStatus
+from src.utils import *
+from src.action_handle_blogger import _interact_with_user, _scroll_to_bottom
+from src.filter import Filter
+
+
+def handle_hashtags(device,
+                   hashtag,
+                   likes_count,
+                   follow_percentage,
+                   storage,
+                   profile_filter,
+                   on_like,
+                   on_interaction):
+    interaction = partial(_interact_with_user,
+                          likes_count=likes_count,
+                          follow_percentage=follow_percentage,
+                          on_like=on_like,
+                          profile_filter=profile_filter)
+
+    if not _open_post_hashtags(device, hashtag):
+        return
+    _iterate_over_posts(device, interaction, storage, on_interaction)
+
+def _open_post_hashtags(device, hashtag):
+    if hashtag is None:
+        print("Enter own hashtag search")
+        tab_bar = device(resourceId='com.instagram.android:id/tab_bar', className='android.widget.LinearLayout')
+        search_button = tab_bar.child(index=1)
+
+        # Two clicks to reset tab content
+        search_button.click.wait()
+    else:
+        print("Press search")
+        tab_bar = device(resourceId='com.instagram.android:id/tab_bar', className='android.widget.LinearLayout')
+        search_button = tab_bar.child(index=1)
+
+        # Two clicks to reset tab content
+        search_button.click.wait()
+        search_button.click.wait()
+
+        print("Open hashtag #" + hashtag)
+        search_edit_text = device(resourceId='com.instagram.android:id/action_bar_search_edit_text',
+                                  className='android.widget.EditText')
+        search_edit_text.set_text(hashtag)
+
+        search_hashtag_button = device(resourceId='com.instagram.android:id/fixed_tabbar_view',
+                            className='android.widget.FrameLayout').child(index=2)
+        search_hashtag_button.click.wait()
+
+        device.wait.idle()
+        hashtag_view = device(resourceId='com.instagram.android:id/row_hashtag_textview_tag_name',
+                               className='android.widget.TextView',
+                               text="#" + hashtag)
+
+        if not hashtag_view.exists:
+            print_timeless(COLOR_FAIL + "Cannot find hashtag #" + hashtag + ", abort." + COLOR_ENDC)
+            return False
+
+        hashtag_view.click.wait()
+
+        print("Open #" + hashtag + " recent posts")
+        hashtag_type = device(resourceId='com.instagram.android:id/tab_layout',

```

Comment:
Does it open "Recent" tab? Maybe `hashtag_type_recent` would be better then

Reply:
Yes it does open the recent tab versus the top tag